### PR TITLE
fix(env): support for uppercase reference to get ENV and SECRET

### LINKED ISSUE #11141: bug: unable to setup ssls with env reference
### Current Behavior

I'm struggling with adding certificate with /ssls admin endpoint using env reference.
```
curl --location --request PUT 'http://127.0.0.1:9180/apisix/admin/ssls' \
--header 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' \
--header 'Content-Type: application/json' \
--data '{
    "id":  "some_id",
    "type": "server",
    "snis": ["localhost"],
    "cert": "$ENV://APISIX_ENV_CERT",
    "key": "$ENV://APISIX_ENV_KEY",
    "ssl_protocols": ["TLSv1.2", "TLSv1.3"]
  }'
```
I'm always getting this, no matter what I put inside cert/key fields unless they are proper certs.
```
{
    "error_msg": "invalid configuration: property \"key\" validation failed: value should match only one schema, but matches none"
}
```
Accordingly to [admin api documentation](https://apisix.apache.org/docs/apisix/admin-api/#request-body-parameters-4) those fields support those env references, but it seems it's not the case. Note: I'm using such env reference in key-auth plugin and it works just fine, but here it seems like it doesn't expect any other pattern than cert, and message/logs are not helpful.

### Expected Behavior

It's possible to setup ssls and provide cert location as env reference.

### Error Logs

_No response_

### Steps to Reproduce

Send following request to admin-api:
```
curl --location --request PUT 'http://127.0.0.1:9180/apisix/admin/ssls' \
--header 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' \
--header 'Content-Type: application/json' \
--data '{
    "id":  "some_id",
    "type": "server",
    "snis": ["localhost"],
    "cert": "$ENV://APISIX_ENV_CERT",
    "key": "$ENV://APISIX_ENV_KEY",
    "ssl_protocols": ["TLSv1.2", "TLSv1.3"]
  }'
```

### Environment

- APISIX version (run `apisix version`): 3.7.0
- Operating system (run `uname -a`): Linux 4091912aa453 6.6.16-linuxkit #1 SMP Fri Feb 16 11:54:02 UTC 2024 x86_64 GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): n/a
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): 3.5.9
- APISIX Dashboard version, if relevant: n/a
- Plugin runner version, for issues related to plugin runners: n/a
- LuaRocks version, for installation issues (run `luarocks --version`): n/a

### PR DESCRIPTION
**Pull request by LinkinStars**
### Description

Fixes #11141

```
"cert": "$ENV://APISIX_ENV_CERT",
"key": "$ENV://APISIX_ENV_KEY",
```

Only lowercase is matched in the 'cert' and 'key' parameters. To maintain consistency, match the uppercase as well. 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by shreemaan-abhishek** [comment]
@LinkinStars please add test cases as well

File: t/router/radixtree-sni2.t [hunk 1]
```diff
@@ -820,3 +820,32 @@ location /t {
...
+            if code >= 300 then
+                ngx.status = code
+                return ngx.say(body)
+            end
+
+            ngx.say("passed")
+        }
+    }
+--- response_body
+passed
```
Comment (shreemaan-abhishek) [inline t/router/radixtree-sni2.t:851]: this test proves that using UPPERCASE works, now please add another test that shows this SSL resource in action (and that the env vars are being resolved)

**Comment by shreemaan-abhishek** [comment]
also, please fix the code lint problem

**Comment by shreemaan-abhishek** [comment]
@LinkinStars, okay. Please rebase with master for the fix for CI failures.

**Comment by shreemaan-abhishek** [comment]
@LinkinStars I think using lowercase "`$env://`" and uppercase "`$SECRET://`" might still not work as the code hardcodes the case for sanity checks:

Hardcoding for secret reference:
https://github.com/shreemaan-abhishek/apisix/blob/07b3cc3f63db8fc4ae3706baa324a76ee3049338/apisix/secret.lua#L34
https://github.com/shreemaan-abhishek/apisix/blob/07b3cc3f63db8fc4ae3706baa324a76ee3049338/apisix/secret.lua#L121

Hardcoding for env reference:
https://github.com/shreemaan-abhishek/apisix/blob/07b3cc3f63db8fc4ae3706baa324a76ee3049338/apisix/core/env.lua#L30
https://github.com/shreemaan-abhishek/apisix/blob/07b3cc3f63db8fc4ae3706baa324a76ee3049338/apisix/core/env.lua#L70

If you were to write test cases covering the above situation it would fail. What could be a good solution for this?

**Comment by shreemaan-abhishek** [comment]
let's only support uppercase for ENV and leave secret as it is. Really appreciate the detailed explanation 🙏🏼

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by Neilblaze — APPROVED** [review]
LGTM!

**Comment by shreemaan-abhishek** [comment]
@LinkinStars please resolve the conflicts.

**Review by shreemaan-abhishek — COMMENTED** [review]
@LinkinStars please resolve the conflicts.

**Pull request closed without merging**